### PR TITLE
test(pkg): reproduce not finding the patch file (`dune fmt`)

### PR DESCRIPTION
**Pull request by moyodiallo**
It reproduce https://github.com/ocaml/dune/issues/10903.

### CODE REVIEW

File: test/blackbox-tests/test-cases/pkg/ocamlformat/patch-extra-files.t [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  > EOF
+  $ cat > ocamlformat.ml <<EOF
+  > let () = print_string "Hello"
+  > EOF
+  $ cat > dune <<EOF
+  > (executable
+  >  (public_name ocamlformat))
+  > EOF
+  $ cd ..
+  $ tar -czf ocamlformat.tar.gz ocamlformat
```
Comment (Leonidas-from-XIV) [inline test/blackbox-tests/test-cases/pkg/ocamlformat/patch-extra-files.t:21]: Nitpick, but if you don't compress the tarball then you avoid a dependency on `gzip`.

Comment (moyodiallo) [reply test/blackbox-tests/test-cases/pkg/ocamlformat/patch-extra-files.t:21]: Wouldn't that mean ocamlformat directory could not be removed ?

Comment (Leonidas-from-XIV) [reply test/blackbox-tests/test-cases/pkg/ocamlformat/patch-extra-files.t]: I agree, opened #10952.

Comment (moyodiallo) [reply test/blackbox-tests/test-cases/pkg/ocamlformat/patch-extra-files.t]: Thanks.

File: test/blackbox-tests/test-cases/pkg/ocamlformat/patch-extra-files.t [hunk 2]
```diff
@@ -0,0 +1,71 @@
+This is a current bug, 'dune fmt' misses extra files from "dev-tools.locks" directory for the first run.
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/pkg/ocamlformat/patch-extra-files.t:1]: I don't really understand the bug from this description

Comment (moyodiallo) [reply test/blackbox-tests/test-cases/pkg/ocamlformat/patch-extra-files.t:1]: I'll rework it.

Comment (moyodiallo) [reply test/blackbox-tests/test-cases/pkg/ocamlformat/patch-extra-files.t]: I think it could be more comprehensible for now.

**Comment by moyodiallo** [comment]
The suggestions are applied, need another review here.

@Leonidas-from-XIV @rgrinberg

File: test/blackbox-tests/test-cases/pkg/ocamlformat/ocamlformat-patch-extra-files.t [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  > +let version = "0.26.2"
+  >  let () =
+  >    if Sys.file_exists ".ocamlformat-ignore" then
+  >    print_endline "ignoring some files"
+  >  ;;
+  >  let () = print_endline ("formatted with version "^version)
+  > EOF
+
+Make the ocamlformat opam package which uses a patch:
+  $ mkpkg ocamlformat "0.26.2"  <<EOF
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/pkg/ocamlformat/ocamlformat-patch-extra-files.t:32]: Do you actually need the solving step? Can't you write a fake lock directory for ocamlformat?

Comment (moyodiallo) [reply test/blackbox-tests/test-cases/pkg/ocamlformat/ocamlformat-patch-extra-files.t:32]: Yeah we need it. The bug is when the solving step is done during the fisrt run of  `dune fmt` and if you have already the lock directory it won't trigger the bug.

**Review by rgrinberg — APPROVED** [review]

**Pull request merged**